### PR TITLE
Developing inside containers

### PR DESCRIPTION
**Pull request by JohnnieJnr**
# Pull Request Template
/claim https://github.com/daytonaio/content/issues/5
/closed https://github.com/daytonaio/content/issues/5
## Writer's Checklist

- [x] **Follow Writing Structure**
  - [x] I have used more than two H2 subtitles to organize content.
  - [x] Include an introduction with at least two paragraphs before the first H2 section.
  - [x] Use bold, italic, underline, notes, quotes, TLDRs, and key points as needed.
  - [x] Include `code elements` where appropriate.
  - [x] Write the article in Markdown format.

- [x] **Fact-Check**
  - [x] I have verified all facts and data points included in the article.

- [x] **Assets**
  - [x] Save images in the `/assets` folder.
  - [x] Follow naming conventions: `YYYYMMDD_title_of_the_article_IMG_NAME_NO.png`.

- [x] **Interlinking**
  - [x] Use `CTRL+F` to search for relevant keywords on:
    - [Daytona Dotfiles Sitemap](https://www.daytona.io/sitemap-dotfiles.xml)
    - [Daytona Definitions Sitemap](https://www.daytona.io/sitemap-definitions.xml)
  - [x] Interlink keywords using the appropriate URLs.

- [x] **Glossary/Definitions**
  - [x] Identify and mark new terms that could be added to the Definitions page.

- [x] **Review and Edit**
  - [x] Read the article out loud to catch any awkward phrasing.
  - [x] Run the draft through [Grammarly](https://grammarly.com) or a similar tool to correct grammar and spelling errors.
  - [x] Double-check adherence to the style guide and repository guidelines.

### CODE REVIEW

**Comment by nkkko** [comment]
I can't review until you implement the checklist. Also article is missing frontmatter.

**Comment by JohnnieJnr** [comment]
I've reviewed my work and implemented the recommendations.

File: articles/20240819_developing_inside_containers.md [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+
+4. **Consistent Development Experience:** By using VSCode’s Remote - Containers extension, we ensure that our development experience is consistent across different platforms, making it easier for new developers to onboard and start contributing to projects.
+
+### Conclusion
+
+Developing inside containers with VSCode is a powerful way to ensure consistency and portability in your development workflow. By following the steps outlined in this guide, you can set up a containerized development environment that suits your needs, avoid common pitfalls, and take inspiration from how Daytona does it to streamline your process. Happy coding!
+
+---
+
+This article should provide a comprehensive, beginner-friendly guide to developing inside containers with VSCode, complete with examples and special instructions to help new developers avoid common issues.
```
Comment (nkkko) [inline articles/20240819_developing_inside_containers.md:290]: Mate what is this? 🤣

**Review by nkkko — CHANGES_REQUESTED** [review]
In general article is well outlined, please invest more time in editing and reviewing, make sure all code snippets make sense and are correct and usable. You will be assigned to this issue.

Read the readme and confirm the checklist!!!

Thanks!

File: articles/20240819_developing_inside_containers.md (hunk 1 above)
Comment (JohnnieJnr) [reply articles/20240819_developing_inside_containers.md:290]: I wrote that to be the header but I'll take it out🤣

File: articles/20240819_developing_inside_containers.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+## TL;DR
+
+Containerization packages your app and its dependencies into a single, portable unit that runs consistently across different environments. This guide walks you through setting up a development environment inside a container using VSCode, covering basics, pitfalls to avoid, and Daytona's approach.
+
+## A Beginner's Guide to Developing Inside Containers with VSCode
+
+### Introduction
+
+Containerization is a game-changer for development, ensuring your app runs the same everywhere. If you're new to this, don't worry—VSCode makes it simple. In this guide, we'll cover setting up a containerized development environment in VSCode, highlight common pitfalls, and share how Daytona does it.
```
Comment (nkkko) [inline articles/20240819_developing_inside_containers.md:17]: avoid constructs such as game-changer

File: articles/20240819_developing_inside_containers.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+[Docker](https://www.daytona.io/definitions/d/docker) is essential for creating and managing containers. Install it from the [Docker website](https://www.docker.com/products/docker-desktop).
+
+After installation, confirm it's working by running:
+
+```bash
+docker --version
+```
+
+#### Step 2: Install VSCode
```
Comment (nkkko) [inline articles/20240819_developing_inside_containers.md:42]: be consistent in using names

**Review by nkkko — CHANGES_REQUESTED** [review]
please write from external contributor POV, do not use we or our.

**Comment by JohnnieJnr** [comment]
fixed all lint issues

**Comment by JohnnieJnr** [comment]
The current errors with Content Compliance Check has to do with the YYYYMMDD_article_template.md file

**Review by nkkko — APPROVED** [review]

**Comment by nkkko** [comment]
Thanks @JohnnieJnr 🥇

**Comment by nkkko** [comment]
Hey @JohnnieJnr please resolve the conflict so that I can merge this.

**Comment by JohnnieJnr** [comment]
@nkkko merge conflict has been resolved.

**Pull request merged**